### DISCUSSION TITLE
B2 uploads stalling and failing to retry on 500 errors

### DISCUSSION THREAD

**Discussion by rhettk** [iterate-ch/cyberduck#14848] [Q&A]
Hi

this latest client is experiencing considerable difficulty uploading 8GB files to B2 using B2 API.

Uploads are stalling - such that app suggests it is uploading, when task manager confirms no network activity whatsoever.  Close app; and it won't re-load unless you kill Cyberduck process in task manager.

I've also found that multiple file uploads at once can fail with a 500 error from B2 but Cyberduck doesn't automatically retry (as recommended by B2).

**Comment by dkocher** [comment]
Please make sure you have set connection retry enabled in _Preferences → Connection → Timeouts → Repeat failed networking tasks_.

**Comment by dkocher** [comment]
Working on #14839 I am currently testing with B2 using large uploads as well. I am unable to reproduce the `500` error responses but always run in to socket timeout errors for uploads of parts of a large file upload when waiting for the response of `b2_upload_part` [^1] requests. Maybe @metadaddy has some input on this?

[^1]: https://www.backblaze.com/b2/docs/b2_upload_part.html

**Reply by dkocher** [reply]
This seems to be related to the particular part size of `104857600` used by default.

**Reply by dkocher** [reply]
Opened issue #14856.

**Reply by dkocher** [reply]
With the part size for large file uploads now following recommendations of the API you should see reduced error rates with the current snapshot build. See [Preferences → Update](https://docs.cyberduck.io/cyberduck/preferences/#update) to switch the update feed.
